pypeman/plugins/base.py

```diff
@@ -0,0 +1,137 @@
+"""
+Base class for a pypeman plugin
+
+All interfaces are specified nothing is implemented
+"""
+
+import logging
+
+from inspect import iscoroutinefunction
+
+
+logger = logging.getLogger(__name__)
+
+
+class BasePlugin:
+    (INITIALIZED, READY, STARTING, STARTED, STOPPING, STOPPED, DESTROYED) = \
+        states = range(7)
+
+    STATE_NAMES = (
+        "INITIALIZED", "READY", "STARTING", "STARTED",
+        "STOPPING", "STOPPED", "DESTROYED",
+        )
+
+    def __init__(self):
+        """
+        Early initialisation before any pypeman endpoints and channels are created
+        """
+        self.name = __name__ + str(self.__class__)
+        logger.debug("Initialized plugin %s", self.name)
+        self.do_init()
+        self.status = self.INITIALIZED
+        self.loop = None
+        self.to_start = []
+        self.start_results = []
+
+    def do_init(self):
```

Comment:
I find more elegant that the user must overide `init(self)` method instead of `do_init(self)`.
Is it possible to swap them ?

Reply:
will look into it

Reply:
Reimplemented with `__init__()` only.

Will adapt existing plugin to use `__init__() and  `super()`

Done in a2bf086ab9c7b147f1de65cbf20ce21710ae8681